src/native/fs/fs_napi.cpp

```diff
@@ -1058,7 +1064,11 @@ struct Readfile : public FSWorker
         }
 
         _len = _stat_res.st_size;
-        _data = new uint8_t[_len];
+        int r = posix_memalign((void**)&_data, DIO_BUFFER_MEMALIGN, _len);
+        if (r || !_data) {
```

Comment:
I don't know if we need the ability to read 0 sized file or not but we do then I think we would error out here.

Quoting `posix_memalign` man page:
> If size is 0, then posix_memalign() returns either NULL, or a unique pointer value that can later be successfully passed to [free](https://linux.die.net/man/3/free)(3).

Reply:
oh, so I guess I should only check the return code, and not the buffer pointer, is that what you meant?

Reply:
@utkarsh-pro see updated version